frontend/src/app/components/BookingHistory.tsx

```diff
@@ -0,0 +1,180 @@
+import { Input, Button, Chip, Pagination } from "@nextui-org/react";
+import React from "react";
+import SearchIcon from "./svgs/SearchIcon";
+
+const dummyData = [
+  {
+    name: "Seaside Cottage",
+    media: "https://example.com/seaside_cottage_image",
+    checkinDate: "April 10, 2024",
+    checkoutDate: "April 15, 2024",
+    bill: 750.0,
+    status: "Upcoming"
+  },
+  {
+    name: "Mountain Cabin",
+    media: "https://example.com/mountain_cabin_image",
+    checkinDate: "May 5, 2024",
+    checkoutDate: "May 10, 2024",
+    bill: 600.0,
+    status: "Done"
+  },
+  {
+    name: "City Apartment",
+    media: "https://example.com/city_apartment_image",
+    checkinDate: "June 20, 2024",
+    checkoutDate: "June 25, 2024",
+    bill: 900.0,
+    status: "Pending"
+  },
+  {
+    name: "Magical Location",
+    media: "https://example.com/city_apartment_image",
+    checkinDate: "June 20, 2024",
+    checkoutDate: "June 25, 2024",
+    bill: 900.0,
+    status: "Cancelled"
+  }
+];
+
+const BookingHistory: React.FC = () => {
+  const getStatusColor = (status: string): string => {
+    switch (status) {
+      case "Upcoming":
+        return "bg-blue-300 text-indigo-500";
+      case "Cancelled":
+        return "bg-danger-200 text-red-500";
+      case "Done":
+        return "bg-success-200 text-green-500";
+      case "Pending":
+        return "bg-warning-200 text-yellow-500";
+      default:
+        return "bg-gray-500 text-white";
+    }
+  };
+
+  const getAction = (status: string): JSX.Element => {
```

Comment:
I think this should be a separate component, so we can reuse it for admin stuff in the future.